### PR TITLE
Add Alluxio FUSE SDK end-to-end testing

### PR DESCRIPTION
**Pull request by LuQQiu**
### What changes are proposed in this pull request?

This changes depends on https://github.com/Alluxio/alluxio/pull/16478

### Why are the changes needed?

To make sure FUSE works with UFS

### Does this PR introduce any user facing changes?

NA

### CODE REVIEW

File: integration/fuse/src/main/java/alluxio/fuse/AlluxioJniFuseFileSystem.java [hunk 1]
```diff
@@ -217,41 +217,45 @@ private int getattrInternal(String path, FileStat stat) {
...
+        Optional<URIStatus> completedFileStatus = AlluxioFuseUtils
+            .waitForFileCompleted(mFileSystem, uri);
+        completedFileStatus.ifPresent(uriStatus
+            -> AlluxioFuseUtils.updateStatSize(stat, uriStatus.getLength()));
+        if (!completedFileStatus.isPresent()) {
+          LOG.error("File {} is not completed, cannot get accurate file length", path);
         }
       }
+
+      if (!status.isPresent() && !hasWriteStream) {
```
Comment (LuQQiu) [inline integration/fuse/src/main/java/alluxio/fuse/AlluxioJniFuseFileSystem.java:251]: Fix the difference between Alluxio and UFS that
UFS files are invisible until finished written (cannot get file status).
Need to check current write streams to understand whether the UFS file is being written by the current FUSE.

File: integration/fuse/src/main/java/alluxio/fuse/AlluxioJniFuseFileSystem.java [hunk 2]
```diff
@@ -547,37 +554,43 @@ private int truncateInternal(String path, long size) {
...
+        return -ErrorCodes.EOPNOTSUPP();
       }
-      LOG.error("Failed to truncate file {}({} bytes) to {} bytes: not supported.",
-          path, fileLen, size);
+
+      LOG.error("Failed to truncate file {} to {} bytes: "
+              + "file is being written by other Fuse applications or Alluxio APIs.",
+          path, size);
+      return -ErrorCodes.EOPNOTSUPP();
+    } catch (UnimplementedRuntimeException e) {
```
Comment (LuQQiu) [inline integration/fuse/src/main/java/alluxio/fuse/AlluxioJniFuseFileSystem.java:589]: Will have future PRs transfer Alluxio related exception to FUSE related exception

**Comment by LuQQiu** [comment]
@jja725 @yyongycy @ssz1997 PTAL, thanks!

File: tests/src/test/java/alluxio/client/fuse/JNIFuseIntegrationTest.java [hunk 3]
```diff
@@ -45,7 +45,7 @@ public void configure() {
 
   @Override
   public void mountFuse(FileSystemContext context,
-      FileSystem fileSystem, String mountPoint, String alluxioRoot) {
+                        FileSystem fileSystem, String mountPoint, String alluxioRoot) {
```
Comment (yyongycy) [inline tests/src/test/java/alluxio/client/fuse/JNIFuseIntegrationTest.java:48]: indent

File: integration/fuse/bin/alluxio-fuse-sdk [hunk 4]
```diff
@@ -187,7 +187,7 @@ launch_fuse_process() {
        alluxio.fuse.AlluxioFuse -m ${mount_point} -u ${ufs_address} ${mount_options}"
 
   if [[ "${foreground}" == 'true' ]]; then
-    exec "${cmd}"
+    exec ${cmd}
```
Comment (yyongycy) [inline integration/fuse/bin/alluxio-fuse-sdk:190]: is the double quote removed by intention?

File: integration/fuse/src/main/java/alluxio/fuse/AlluxioJniFuseFileSystem.java (hunk 1 above)
Comment (yyongycy) [reply integration/fuse/src/main/java/alluxio/fuse/AlluxioJniFuseFileSystem.java:251]: what if the user creates a same file before completion?

File: integration/fuse/bin/alluxio-fuse-sdk (hunk 4 above)
Comment (LuQQiu) [reply integration/fuse/bin/alluxio-fuse-sdk:190]: Yeah, with double quote, the command will fail which is strange, saying that no such command

**Review by yyongycy — APPROVED** [review]
LGTM

**Comment by LuQQiu** [comment]
alluxio-bot, merge this please

File: integration/fuse/src/main/java/alluxio/fuse/AlluxioJniFuseFileSystem.java (hunk 1 above)
Comment (LuQQiu) [reply integration/fuse/src/main/java/alluxio/fuse/AlluxioJniFuseFileSystem.java:251]: Sorry that miss this question.
Multiple scenarios may happen:
1. User wants to create the same file through the same FUSE. the user needs to get a write lock to be able to write to the same file and will wait for the previous one to finish.
2. User creates the same file outside this FUSE, through other Alluxio Fuse or other methods. In this case, i assume only one write will success. If the current FUSE, then no problem, if other ways, the current fuse write will be errored out and FUSE kernel will cause release to close the write stream and unlock.

**Pull request merged**